src/analysis/asymptotics.lean

```diff
@@ -0,0 +1,438 @@
+end
+
+theorem is_littleo.sub {f₁ f₂ : α → β} {g : α → γ} {l : filter α}
+    (h₁ : is_littleo f₁ g l) (h₂ : is_littleo f₂ g l) :
+  is_littleo (λ x, f₁ x - f₂ x) g l :=
+h₁.add (is_littleo_neg_left.mpr h₂)
+
+end
+
+section
+variables [normed_field β] [has_norm γ]
+
+theorem is_bigo_mul_left {f : α → β} {g : α → γ} {l : filter α} {c : β} (hc : c ≠ 0) :
+  is_bigo (λ x, c * f x) g l ↔ is_bigo f g l :=
+begin
+  have cne0 : ∥c∥ ≠ 0, by rw [ne, norm_eq_zero]; exact hc,
+  have cpos : ∥c∥ > 0, from lt_of_le_of_ne (norm_nonneg _) (ne.symm cne0),
+  split,
+  { rintro ⟨c', h'⟩,
+    use (∥c∥⁻¹ * c'),
+    convert h', ext x, dsimp,
+    rw [normed_field.norm_mul, mul_assoc, mul_comm, mul_comm ∥c∥],
+    apply le_div_iff cpos },
+  rintro ⟨c', h'⟩,
+  use (∥c∥ * c'),
+  filter_upwards [h'], dsimp,
+  intros x h₀,
+  rw [normed_field.norm_mul, mul_assoc],
+  exact mul_le_mul_of_nonneg_left h₀ (norm_nonneg _)
+end
+
+theorem is_littleo_mul_left {f : α → β} {g : α → γ} {l : filter α} {c : β} (hc : c ≠ 0) :
+  is_littleo (λ x, c * f x) g l ↔ is_littleo f g l :=
+begin
+  have cne0 : ∥c∥ ≠ 0, by rw [ne, norm_eq_zero]; exact hc,
+  have cpos : ∥c∥ > 0, from lt_of_le_of_ne (norm_nonneg _) (ne.symm cne0),
+  split,
+  { intros h' c',
+    convert h' (c' * ∥c∥⁻¹), ext x, dsimp,
+    rw [normed_field.norm_mul, mul_assoc, ←mul_assoc ∥c∥⁻¹, inv_mul_cancel cne0, one_mul] },
+  intros h' c',
+  convert h' (c' * ∥c∥), ext x, dsimp,
+  rw [normed_field.norm_mul, mul_assoc]
+end
+
+end
+
+section
+variables [has_norm β] [normed_field γ]
+
+theorem is_bigo_mul_right {f : α → β} {g : α → γ} {l : filter α} {c : γ} (hc : c ≠ 0) :
+  is_bigo f (λ x, c * g x) l ↔ is_bigo f g l :=
+begin
+  split,
+  { rintro ⟨c', h'⟩,
+    use (c' * ∥c∥),
+    convert h', ext x, dsimp,
+    rw [normed_field.norm_mul, mul_assoc] },
+  have cne0 : ∥c∥ ≠ 0, by rw [ne, norm_eq_zero]; exact hc,
+  rintro ⟨c', h'⟩,
+  use (c' * ∥c∥⁻¹),
+  convert h', ext x, dsimp,
+  rw [normed_field.norm_mul, mul_assoc, ←mul_assoc ∥c∥⁻¹, inv_mul_cancel cne0, one_mul]
+end
+
+theorem is_littleo_mul_right {f : α → β} {g : α → γ} {l : filter α} {c : γ} (hc : c ≠ 0) :
+  is_littleo f (λ x, c * g x) l ↔ is_littleo f g l :=
+begin
+  have cne0 : ∥c∥ ≠ 0, by rw [ne, norm_eq_zero]; exact hc,
+  have cpos : ∥c∥ > 0, from lt_of_le_of_ne (norm_nonneg _) (ne.symm cne0),
+  split,
+  { intros h' c',
+    filter_upwards [h' (∥c∥ * c')], dsimp,
+    intro x, rw [normed_field.norm_mul, mul_assoc],
+    intro h₀, exact le_of_mul_le_mul_left h₀ cpos },
+  intros h' c',
+  convert h' (∥c∥⁻¹ * c'), ext x, dsimp,
+  symmetry,
+  rw [normed_field.norm_mul, mul_assoc, mul_comm],
+  apply div_le_iff' cpos
+end
+
+end
+
+/-
+Note: the method in the next two sections can be used for the integers, or more generally any normed
+ring satisfying `∥a * b∥ = ∥a∥ * ∥b∥`. It is not clear whether it is worth stating it in that
+generality.
+-/
+
+section
+variables {k : Type*} [normed_field k] [normed_space k β] [has_norm γ]
```

Comment:
Can the field be `K` instead of `k`? I won't insist on greek but you use `k` as a number in some theorems and as a type in others, and it's a bit disorienting.